### PR TITLE
Add withSafeAreaConsumer HOC

### PR DESCRIPTION
**Pull request by jdnichollsc**
## Summary
It's a good option to inject the insets of SafeAreaConsumer as props from React class components

Thanks in advance!

### CODE REVIEW

**Comment by jdnichollsc** [comment]
Check please 👍

**Comment by jdnichollsc** [comment]
Hello @janicduplessis, how are you?
What do you think about this little change?
Thanks!

**Comment by jdnichollsc** [comment]
Is this repo being maintained? :)

File: src/index.tsx [hunk 1]
```diff
@@ -41,6 +41,16 @@ const styles = StyleSheet.create({
 
 export const SafeAreaConsumer = SafeAreaContext.Consumer;
 
+export function withSafeAreaConsumer<T>(
```
Comment (janicduplessis) [inline src/index.tsx:44]:
```suggestion
export function withSafeAreaInsets<T>(
```

Comment (jdnichollsc) [reply src/index.tsx:44]: perfect, let me do that!

Comment (jdnichollsc) [reply src/index.tsx]: Done! please let me know what you think!

**Comment by janicduplessis** [comment]
👍  Looks good, thanks!

**Comment by jdnichollsc** [comment]
thanks to you!

**Pull request merged**